src/tmpfiles/tmpfiles.c

```diff
@@ -181,33 +191,81 @@ STATIC_DESTRUCTOR_REGISTER(arg_root, freep);
+}
+
+static const SpecifierForward specifier_table_data[] = {
+        { { 'm', specifier_machine_id_safe }, SPECTABLE_PREFIXARG },
+        { { 'b', specifier_boot_id },         SPECTABLE_RUNTIME },
+        { { 'H', specifier_host_name },       SPECTABLE_RUNTIME },
+        { { 'l', specifier_short_host_name }, SPECTABLE_RUNTIME },
+        { { 'v', specifier_kernel_release },  SPECTABLE_RUNTIME },
+        { { 'a', specifier_architecture },    SPECTABLE_RUNTIME },
+        { { 'o', specifier_os_id },           SPECTABLE_PREFIXARG },
+        { { 'w', specifier_os_version_id },   SPECTABLE_PREFIXARG },
+        { { 'B', specifier_os_build_id },     SPECTABLE_PREFIXARG },
+        { { 'W', specifier_os_variant_id },   SPECTABLE_PREFIXARG },
+
+        { { 'g', specifier_group_name },      SPECTABLE_USERDB },
+        { { 'G', specifier_group_id },        SPECTABLE_USERDB },
+        { { 'U', specifier_user_id },         SPECTABLE_USERDB },
+        { { 'u', specifier_user_name },       SPECTABLE_USERDB },
+        { { 'h', specifier_user_home },       SPECTABLE_USERDB },
+
+        { { 't', specifier_directory, UINT_TO_PTR(DIRECTORY_RUNTIME) }, SPECTABLE_DIRECTORY },
+        { { 'S', specifier_directory, UINT_TO_PTR(DIRECTORY_STATE) },   SPECTABLE_DIRECTORY },
+        { { 'C', specifier_directory, UINT_TO_PTR(DIRECTORY_CACHE) },   SPECTABLE_DIRECTORY },
+        { { 'L', specifier_directory, UINT_TO_PTR(DIRECTORY_LOGS) },    SPECTABLE_DIRECTORY },
+        { { 'T', specifier_tmp_dir,     "/tmp" },     SPECTABLE_STRING },
+        { { 'V', specifier_var_tmp_dir, "/var/tmp" }, SPECTABLE_STRING },
 };
 
+static Specifier specifier_table[ELEMENTSOF(specifier_table_data) + 1];
+
+static void setup_specifier_table(const char *prefix) {
+        int r, i;
+        int prohibit_runtime = 0, is_online = prefix == NULL;
+        (void) is_online;
+
+        r = getenv_bool("SYSTEMD_SPECIFIER_PROHIBIT_RUNTIME");
+        if (r > 0)
+                prohibit_runtime = r;
+
+        for (i = 0; i < (int) ELEMENTSOF(specifier_table_data); ++i) {
+                SpectableFlag flag = specifier_table_data[i].flag;
+                Specifier spec = specifier_table_data[i].specifier;
+
+                if (prohibit_runtime) {
+                        switch (flag) {
+                        case SPECTABLE_PREFIXARG:
+                                spec.data = prefix;
+                                break;
+                        case SPECTABLE_STRING:
+                                spec.lookup = specifier_string;
+                                break;
+                        case SPECTABLE_RUNTIME:
+                        case SPECTABLE_USERDB:
+                                spec.lookup = disabled_specifier;
+                                break;
+                        default:
+                                break;
+                        }
+                } else {
+                        // remove arguments only used for offline
```

Comment:
as per coding style, we use `//` comments for `FIXME` style stuff only, and `/* */` for explanatory comments.